php/traits/trait-params.php

```diff
@@ -53,10 +53,15 @@ protected function set_param_array( $parts, $param, $value ) {
 			return $new;
 		}
 
-		if ( '' === $key ) {
-			$new[] = $value;
+		if ( is_array( $new ) ) {
+			if ( '' === $key ) {
+				$new[] = $value;
+			} else {
+				$new[ $key ] = $value;
+			}
 		} else {
-			$new[ $key ] = $value;
+			// There's likely a problem here, as we're setting an array key on a non-array.
+			$new = array();
```

Comment:
<!-- potential concern, ask for clarification or investigation -->
**❓question**
I'm approving this but want to check that once `$new` is an array, you don't want to set the key/value on it?

Reply:
That's correct. If `$param` and hence `$new` is a string, then that three is broken, and we are better off ignoring it. Upon a new save, it should be rebuilt accordingly. It's as if a new setting was introduced.